dataset_generation/parse_textbook.py

```diff
@@ -0,0 +1,84 @@
+from argparse import ArgumentParser
+import json
+import os
+import fitz # if missing, pip install pymupdf
```

Comment:
nit :: could you add a requirements.txt for this? or include a list of all the dependencies you need on the `./dataset_generation/README.md` file?

Reply:
will do, I'll make another PR with a README and requirements.txt